presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/StateCompiler.java

```diff
@@ -535,15 +522,17 @@ private static FieldDefinition generateGroupedField(ClassDefinition definition,
     /**
      * Enumerates all the fields in this state interface.
      *
-     * @param clazz a subclass of AccumulatorState
+     * @param stateClass a wrapper of a subclass of AccumulatorState with field types
```

Comment:
Remove this line (or move this line to the constructor of `StateClass`).

In `StateClass` constructor, validate that `clazz` is a subclass of AccumulatorState.

In fact, this check can be a compile-time check. But I acknowledge that such a refactor would touch too many lines that are being changed in this pull request. Add a TODO and do it after this pull request is merged.